plugin/manager.go

```diff
@@ -263,12 +268,23 @@ func parsePluginFromDir(path string) (string, string, error) {
 			}
 			foundPluginExecutableFile = true
 			pluginExecutableFile = p
+			pluginName = candidatePluginName
 		}
 		return nil
 	}); err != nil {
 		return "", "", err
 	}
 	if !foundPluginExecutableFile {
+		// if no executable file was found, but there's one and only one
+		// potential candidate, try install the candidate
+		if len(filesWithValidNameFormat) == 1 {
+			candidate := filesWithValidNameFormat[0]
+			if err := setExecutable(candidate); err != nil {
+				return "", "", fmt.Errorf("no plugin executable file was found: %w", err)
+			}
+			logger.Warnf("Found candidate plugin executable file %q without executable permission. Setting user executable bit and trying to install.", filepath.Base(candidate))
+			return candidate, candidatePluginName, nil
+		}
 		return "", "", errors.New("no plugin executable file was found")
```

Comment:
What happens if there are two files without executable permission: notation-plugin-a and notation-plugin-b? In this case, what error should be returned?

Reply:
@JeyJeyGao In this case, the install would fail with error: `failed to read plugin from input directory: no plugin executable file was found`

Reply:
The corresponding test is covered at here: https://github.com/notaryproject/notation-go/blob/3d8621dbf47107d04186ebf956b0b837783834c7/plugin/manager_test.go#L504